Makefile

```diff
@@ -1,19 +1,26 @@
 #!/usr/bin/make
 
+TOX_VER = $(shell tox --version 2>/dev/null || echo "0.0")
+TOX_VER_MAJOR := $(shell echo $(TOX_VER) | cut -f1 -d.)
+TOX_VER_MINOR := $(shell echo $(TOX_VER) | cut -f2 -d.)
+TOX_GT_1_8 := $(shell [ $(TOX_VER_MAJOR) -gt 1 -o \( $(TOX_VER_MAJOR) -eq 1 -a $(TOX_VER_MINOR) -ge 8 \) ] && echo true)
+
 all: lint unit_test
 
 
 .PHONY: clean
 clean:
 	@rm -rf .tox
 
-.PHONY: apt_prereqs
-apt_prereqs:
-	@# Need tox, but don't install the apt version unless we have to (don't want to conflict with pip)
-	@which tox >/dev/null || (sudo apt-get install -y python-pip && sudo pip install tox)
+.PHONY: tox_prereqs
+tox_prereqs:
+ifneq ($(TOX_GT_1_8),true)
+        @echo "Pip installing tox"
+        sudo apt-get install -y python-pip && sudo pip install -U 'tox>=1.8'
```

Comment:
could we just, for now, install to --user? even though I'd prefer a virtualenv for tox. If I run the Makefile in this layer, it's only going to flake and charm proof, none of these should run scripts that do a sudo install, so your argument that because tests do this should don't follow through.